File: frontend/src/types/game.ts
export type PowerBracket = 'precon' | 'casual' | 'optimized' | 'cedh';

export type GameMode = 'rules_enforced' | 'manual';

export type GameStatus = 'waiting' | 'in_progress' | 'completed';

export type PlayerStatus = 'pending' | 'accepted' | 'rejected';

export interface DeckInfo {
  id: number;
  name: string;
  commander_name?: string;
  commander_image_uris?: {
    small?: string;
    normal?: string;
    large?: string;
    art_crop?: string;
    border_crop?: string;
  };
}

export interface GameRoomPlayer {
  id: number;
  user_id: number;
  username: string;
  status: PlayerStatus;
  is_host: boolean;
  deck_id?: number;
  deck?: DeckInfo;
  joined_at: string;
}

export interface GameRoom {
  id: number;
  name: string;
  description?: string;
  host_id: number;
  host_username: string;
  invite_code: string;
  is_public: boolean;
  max_players: number;
  power_bracket: PowerBracket;
  game_mode: GameMode;
  status: GameStatus;
  players: GameRoomPlayer[];
  created_at: string;
}

export interface GameRoomListItem {
  id: number;
  name: string;
  description?: string;
  host_username: string;
  is_public: boolean;
  max_players: number;
  current_players: number;
  power_bracket: PowerBracket;
  game_mode: GameMode;
  status: GameStatus;
  created_at: string;
  is_in_game: boolean;
  is_host: boolean;
}

export interface GameRoomCreate {
  name: string;
  description?: string;
  is_public: boolean;
  max_players: number;
  power_bracket: PowerBracket;
  game_mode: GameMode;
}

export const POWER_BRACKET_LABELS: Record<PowerBracket, string> = {
  precon: 'Precon',
  casual: 'Casual',
  optimized: 'Optimized',
  cedh: 'cEDH',
};

export const POWER_BRACKET_COLORS: Record<PowerBracket, string> = {
  precon: 'bg-blue-500/20 text-blue-400 border-blue-500/30',
  casual: 'bg-green-500/20 text-green-400 border-green-500/30',
  optimized: 'bg-orange-500/20 text-orange-400 border-orange-500/30',
  cedh: 'bg-red-500/20 text-red-400 border-red-500/30',
};

export const GAME_MODE_LABELS: Record<GameMode, string> = {
  rules_enforced: 'Rules Enforced',
  manual: 'Manual Mode',
};

export const GAME_MODE_COLORS: Record<GameMode, string> = {
  rules_enforced: 'bg-purple-500/20 text-purple-400 border-purple-500/30',
  manual: 'bg-gray-500/20 text-gray-400 border-gray-500/30',
};

File: backend/app/schemas/game.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional, List
from datetime import datetime
from app.models.game import PowerBracket, GameStatus, PlayerStatus


class DeckInfo(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    commander_name: Optional[str] = None
    commander_image_uris: Optional[dict] = None


class GameRoomCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    description: Optional[str] = Field(None, max_length=500)
    is_public: bool = False
    max_players: int = Field(default=4, ge=2, le=4)
    power_bracket: PowerBracket = PowerBracket.CASUAL


class GameRoomUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    description: Optional[str] = Field(None, max_length=500)
    is_public: Optional[bool] = None
    max_players: Optional[int] = Field(None, ge=2, le=4)
    power_bracket: Optional[PowerBracket] = None


class DeckSelectionRequest(BaseModel):
    deck_id: int


class GameRoomPlayerResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    username: str
    status: PlayerStatus
    is_host: bool
    deck_id: Optional[int] = None
    deck: Optional[DeckInfo] = None
    joined_at: datetime


class GameRoomResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: Optional[str]
    host_id: int
    host_username: str
    invite_code: str
    is_public: bool
    max_players: int
    power_bracket: PowerBracket
    status: GameStatus
    players: List[GameRoomPlayerResponse]
    created_at: datetime


class GameRoomListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: Optional[str]
    host_username: str
    is_public: bool
    max_players: int
    current_players: int
    power_bracket: PowerBracket
    status: GameStatus
    created_at: datetime
    is_in_game: bool = False


class JoinResponse(BaseModel):
    message: str
    game_room: Optional[GameRoomResponse] = None
